tools/castle-editor/code/framedesign.pas

```diff
@@ -2026,6 +2028,24 @@ procedure TDesignFrame.PropertyGridModified(Sender: TObject);
   MarkModified;
 end;
 
+procedure TDesignFrame.PropertyEditorModified(Sender: TObject);
+var
+  Sel: TComponent;
+begin
+  Sel := SelectedComponent;
+  if Sender is TPropertyEditor then
+  begin
+    if Sel <> nil then
+      RecordUndo(Sel.Name + ': change ' + TPropertyEditor(Sender).GetName + ' to ' + TPropertyEditor(Sender).GetValue, ucHigh)
+    else
+      RecordUndo('Change ' + TPropertyEditor(Sender).GetName + ' to ' + TPropertyEditor(Sender).GetValue, ucHigh)
+  end else
+    if Sel <> nil then
```

Comment:
As with all unexpected bugs with Undo system, I'd rather keep things as robust as possible, than to throw an exception in some situation that seems absolutely impossible but still happens on random (like `Sel=nil` that was causing access violation in `PropertyGridModified`).

Reply:
In general I say we have to explain something that """seems absolutely impossible but still happens on random""" :)

I case of `Sel` being `nil` in `PropertyGridModified`, this can perfectly happen in case of multiple components selected.

So we want to be robust (tolerate everything that *can* happen), but we need to reason about "what things can happen". If something is impossible, then it is better to make an exception in this case, and then if someone reports it -- we have a proof "OK, this can happen, in this-and-this situation".

Reply:
Done in 29dd666e4945c30d6cad64625d001bd0854461b6!